recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,157 @@
+        "boost":      False,
+        "valgrind":   False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        prefix  = "Dependencies for "
+        message = " not configured in this conan package, some features will be disabled."
+
+        if self.options.sqlite3:
+            self.requires("sqlite3/3.33.0")
+        # ToDo add the missing dependencies for the backends
+        if self.options.db2:
+            self.output.warn(prefix + "DB2" + message)
+        if self.options.odbc:
+            self.output.warn(prefix + "ODBC" + message)
+        if self.options.oracle:
+            self.output.warn(prefix + "ORACLE" + message)
+        if self.options.firebird:
+            self.output.warn(prefix + "FIREBIRD" + message)
+        if self.options.mysql:
+            self.output.warn(prefix + "MYSQL" + message)
+        if self.options.postgresql:
+            self.output.warn(prefix + "POSTGRESQL" + message)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["SOCI_STATIC"]     = self.options.static
+        self._cmake.definitions["SOCI_CXX11"]      = self.options.cxx11
+        self._cmake.definitions["SOCI_EMPTY"]      = self.options.empty
+        self._cmake.definitions["SOCI_SHARED"]     = self.options.shared
+        self._cmake.definitions["SOCI_TESTS"]      = self.options.tests
+        self._cmake.definitions["WITH_SQLITE3"]    = self.options.sqlite3
+        self._cmake.definitions["WITH_DB2"]        = self.options.db2
+        self._cmake.definitions["WITH_ODBC"]       = self.options.odbc
+        self._cmake.definitions["WITH_ORACLE"]     = self.options.oracle
+        self._cmake.definitions["WITH_FIREBIRD"]   = self.options.firebird
+        self._cmake.definitions["WITH_MYSQL"]      = self.options.mysql
+        self._cmake.definitions["WITH_POSTGRESQL"] = self.options.postgresql
+        self._cmake.definitions["WITH_BOOST"]      = self.options.boost
+        self._cmake.definitions["WITH_VALGRIND"]   = self.options.valgrind
+
+        if self.options.cxx11:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = "11" # ToDo review this. The standard should not be set here
+
+        self._cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder)
+
+        return self._cmake
+
+    def package(self):
```

Comment:
is there a particular reason to not use the `cmake.install()` ?

Reply:
Is run later

Reply:
There is an issue because the install is called in the package info which it not the correct location... it should be in package and you could most likely remove the manual copying